### ISSUE TITLE
False positive for discarded_notification_center_observer rule

### ISSUE THREAD

**Issue by Jumhyn** [realm/SwiftLint#3224]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following code triggers the `discarded_notification_center_observer` rule:

```swift
self.observers.append(
    NotificationCenter.default.addObserver(
        forName: UIApplication.willEnterForegroundNotification,
        object: nil,
        queue: nil
    ) {
        print($0)
    }
)
```

It looks like a similar issue was fixed in #2685, but it didn't cover the case where the call begins on the next line.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
% swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/path/to/test.swift:2:5: warning: Discarded Notification Center Observer Violation: When registering for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. (discarded_notification_center_observer)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version 0.39.1.
* CocoaPods installation.

**Comment by Jumhyn** [comment]
AFAIK this is still an issue... @jpsim could you take a look and reopen if so?
